plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/TableMetadata.java

```diff
@@ -50,6 +50,7 @@
     private final Map<String, String> serdeParameters;
 
     private final Optional<String> externalLocation;
+    private final Optional<String> location;
```

Comment:
Why is this change necessary?

(And should it be in a separate commit?)

Reply:
This is needed to expose the correct `table.storage.location` . Before this change, only the external tables were allowed to expose a different location than the table name within the schema directory.